### PR TITLE
update broken links

### PR DESCRIPTION
**Pull request by mirnawong1**
quickly reviewing some broken links and updating them per this thread [here](https://dbt-labs.slack.com/archives/C03SK9W08BC/p1704473024350309).

materialization and other source codes have moved. this pr updates and replaces github links that started with:
- `https://github.com/dbt-labs/dbt-core/tree/main/core/dbt/adapters/..`
- `https://github.com/dbt-labs/dbt-core/blob/main/core/dbt..`

and replaced it with the permalinks per convo.i did a quick scan to see if there were any other files to update and i didn't see any more.

### CODE REVIEW

File: website/docs/reference/events-logging.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ _For more details about how the eventing system has been implemented in dbt-core
 
 <VersionBlock firstVersion="1.4">
 
-Starting in v1.4, the structure of each event in `dbt-core` is backed by a schema defined using [protocol buffers](https://developers.google.com/protocol-buffers). All schemas are defined in the [`types.proto`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/types.proto) file within the `dbt-core` codebase.
+Starting in v1.4, the structure of each event in `dbt-core` is backed by a schema defined using [protocol buffers](https://developers.google.com/protocol-buffers). All schemas are defined in the [`types.proto`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/events/core_types.proto) file within the `dbt-core` codebase.
```
Comment (mirnawong1) [inline website/docs/reference/events-logging.md]: i think given the context, this is the right link we want to link to right @jtcohen6 ? there's another 'types.proto' file in the `dbt-adapter` repo but it seem like this is the more appropriate one?

File: website/docs/guides/create-new-materializations.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ recently_updated: true
 
 ## Introduction
 
-The model <Term id="materialization">materializations</Term> you're familiar with, `table`, `view`, and `incremental` are implemented as macros in a package that's distributed along with dbt. You can check out the [source code for these materializations](https://github.com/dbt-labs/dbt-core/tree/main/core/dbt/adapters/include/global_project/macros/materializations). If you need to create your own materializations, reading these files is a good place to start. Continue reading below for a deep-dive into dbt materializations.
+The model <Term id="materialization">materializations</Term> you're familiar with, `table`, `view`, and `incremental` are implemented as macros in a package that's distributed along with dbt. You can check out the [source code for these materializations](https://github.com/dbt-labs/dbt-adapters/tree/main/dbt/include/global_project/macros/materializations). If you need to create your own materializations, reading these files is a good place to start. Continue reading below for a deep-dive into dbt materializations.
```
Comment (mirnawong1) [inline website/docs/guides/create-new-materializations.md]:
```suggestion
The model <Term id="materialization">materializations</Term> you're familiar with, `table`, `view`, and `incremental` are implemented as macros in a package that's distributed along with dbt. You can check out the [source code for these materializations](https://github.com/dbt-labs/dbt-adapters/tree/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations). If you need to create your own materializations, reading these files is a good place to start. Continue reading below for a deep-dive into dbt materializations.
```

File: website/docs/guides/debug-schema-names.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ You can also follow along via this video:
 Do a file search to check if you have a macro named `generate_schema_name` in the `macros` directory of your project.
 
 ### You do not have a macro named `generate_schema_name` in your project
-This means that you are using dbt's default implementation of the macro, as defined [here](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/get_custom_name/get_custom_schema.sql#L47C1-L60)
+This means that you are using dbt's default implementation of the macro, as defined [here](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/get_custom_name/get_custom_schema.sql)
```
Comment (mirnawong1) [inline website/docs/guides/debug-schema-names.md]:
```suggestion
This means that you are using dbt's default implementation of the macro, as defined [here](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/get_custom_name/get_custom_schema.sql)
```

File: website/docs/reference/resource-configs/full_refresh.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ Optionally set a resource to always or never full-refresh.
 
 </VersionBlock>
 
-This logic is encoded in the [`should_full_refresh()`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/adapters/include/global_project/macros/materializations/configs.sql#L6) macro.
+This logic is encoded in the [`should_full_refresh()`](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/materializations/configs.sql) macro.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/full_refresh.md]:
```suggestion
This logic is encoded in the [`should_full_refresh()`](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations/configs.sql) macro.
```

File: website/docs/reference/events-logging.md (hunk 1 above)
Comment (mirnawong1) [inline website/docs/reference/events-logging.md]:
```suggestion
Starting in v1.4, the structure of each event in `dbt-core` is backed by a schema defined using [protocol buffers](https://developers.google.com/protocol-buffers). All schemas are defined in the [`types.proto`](https://github.com/dbt-labs/dbt-core/blob/3bf148c443e6b1da394b62e88a08f1d7f1d8ccaa/core/dbt/events/core_types.proto) file within the `dbt-core` codebase.
```

File: website/docs/reference/resource-configs/store_failures.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ Optionally set a test to always or never store its failures in the database.
 - If the `store_failures` config is `none` or omitted, the resource will use the value of the `--store-failures` flag.
 - When true, `store_failures` save all the record(s) that failed the test only if [limit](/reference/resource-configs/limit) is not set or if there are fewer records than the limit. `store_failures` are saved in a new table with the name of the test. By default, `store_failures` use a schema named `dbt_test__audit`, but, you can [configure](/reference/resource-configs/schema#tests) the schema to a different value. 
 
-This logic is encoded in the [`should_store_failures()`](https://github.com/dbt-labs/dbt-core/blob/77632122974b28967221758b4a470d7dfb608ac2/core/dbt/adapters/include/global_project/macros/materializations/configs.sql#L15) macro.
+This logic is encoded in the [`should_store_failures()`](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/materializations/configs.sql#L15) macro.
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/store_failures.md]:
```suggestion
This logic is encoded in the [`should_store_failures()`](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations/configs.sql#L15) macro.
```

File: website/docs/reference/resource-configs/strategy.md [hunk 6]
```diff
@@ -132,8 +132,8 @@ This is a **required configuration**. There is no default value.
 
 ### Advanced: define and use custom snapshot strategy
 Behind the scenes, snapshot strategies are implemented as macros, named `snapshot_<strategy>_strategy`
-* [Source code](https://github.com/dbt-labs/dbt-core/blob/HEAD/core/dbt/adapters/include/global_project/macros/materializations/snapshots/strategies.sql#L52) for the timestamp strategy
-* [Source code](https://github.com/dbt-labs/dbt-core/blob/HEAD/core/dbt/adapters/include/global_project/macros/materializations/snapshots/strategies.sql#L136) for the check strategy
+* [Source code](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L52) for the timestamp strategy
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/strategy.md]:
```suggestion
* [Source code](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L52) for the timestamp strategy
```

File: website/docs/reference/resource-configs/strategy.md [hunk 7]
```diff
@@ -132,8 +132,8 @@ This is a **required configuration**. There is no default value.
...
+* [Source code](https://github.com/dbt-labs/dbt-adapters/blob/main/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L136) for the check strategy
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/strategy.md]:
```suggestion
* [Source code](https://github.com/dbt-labs/dbt-adapters/blob/60005a0a2bd33b61cb65a591bc1604b1b3fd25d5/dbt/include/global_project/macros/materializations/snapshots/strategies.sql#L136) for the check strategy
```

**Review by nghi-ly — APPROVED** [review]
thanks for fixing this!

**Pull request merged**